vllm/sequence.py

```diff
@@ -716,6 +737,33 @@ def token_chunk_size(self) -> int:
         assert self._token_chunk_size is not None
         return self._token_chunk_size
 
+    @property
+    def num_lookahead_slots(self) -> int:
+        return self.state.num_lookahead_slots
+
+    @property
+    def num_steps(self) -> int:
+        return self.state.num_steps
+
+    @property
+    def remaining_steps(self) -> int:
+        return self.state.remaining_steps
+
+    @property
+    def current_step(self) -> int:
+        return self.state.current_step
+
+    @property
+    def is_first_multi_step(self) -> bool:
+        return self.state.current_step == 0
```

Comment:
This looks a bit un-scalable and unclear to me as we are manually exposing all state properties. Maybe explicitly using `seq_group.state.xxx` would be sufficient?